mlx_graphs/utils/array_ops.py

```diff
@@ -45,3 +48,58 @@ def expand(array: mx.array, new_shape: tuple) -> mx.array:
         max(orig_dim, new_dim) for orig_dim, new_dim in zip(orig_shape, new_shape)
     )
     return mx.broadcast_to(array, broadcast_shape)
+
+
+def one_hot(labels: mx.array, num_classes: Optional[int] = None):
+    """
+    Creates one-hot encoded vectors for all elements provided in `labels`.
+
+    Given an array of labels [num_elements,], returns an array with shape
+    [num_elements, num_classes]where each column is an all-zero vector with
+    a one at the index of the label.
+
+    Args:
+        labels: Array with the labels to transform to one-hot encoded vectors.
+        num_classes: Number of labels for the one-hot encoding. By default,
+            ``num_classes`` is set to `max_label + 1`.
+
+    Returns:
+        An array of shape [num_elements, num_classes] with one-hot encoded vectors.
+    """
+    if num_classes is None:
+        num_classes = (labels.max() + 1).item()
+
+    shape = (labels.shape[0], num_classes)
+    one_hot = mx.zeros(shape)
+
+    one_hot[mx.arange(shape[0]), labels.squeeze()] = 1
+    return one_hot
+
+
+def remove_duplicate_directed_edges(edge_index: mx.array) -> mx.array:
+    """
+    Removes the duplicate directed edges in `edge_index`.
+
+    Args:
+        edge_index: The `edge_index` on which remove duplicate edges.
+
+    Returns:
+        An `edge_index` without duplicates.
+    """
+    base_dtype = edge_index.dtype
+    if base_dtype == mx.int64:
+        raise ValueError("Does not support yet `edge_index` with int64 dtype.")
+
+    # TODO: this method shall be updated to full mlx once unique() and structured
+    # arrays are implemented
+    edge_index = np.array(edge_index, np.int32)
+
+    # Convert edge_index to a structured array to leverage np.unique for
+    # multi-column uniqueness
+    dtype = [("node1", edge_index.dtype), ("node2", edge_index.dtype)]
+    structured_edges = np.array(list(zip(edge_index[0], edge_index[1])), dtype=dtype)
+
+    unique_edges = np.unique(structured_edges)
+    unique_edge_index = np.array([unique_edges["node1"], unique_edges["node2"]])
+
+    return mx.array(unique_edge_index, mx.int32)
```

Comment:
This should probably stay in `transformations.py` and the input should be validated.

Reply:
moved in [9b1ec84](https://github.com/mlx-graphs/mlx-graphs/pull/81/commits/9b1ec8490e9eacdee309a149f75197bbebf2f9a4)